yaks/lib/yaks/behaviour/optional_includes.rb

```diff
@@ -0,0 +1,31 @@
+require "rack/utils"
+
+module Yaks
+  module Behaviour
+    module OptionalIncludes
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      module ClassMethods
+        def has_one(name, options = {}) # rubocop:disable Style/PredicateName
+          super name, options.merge(if: ->{include_association?(name)})
+        end
+
+        def has_many(name, options = {}) # rubocop:disable Style/PredicateName
+          super name, options.merge(if: ->{include_association?(name)})
+        end
```

Comment:
With the new implementation proposed by @plexus it's no longer valid to implement it this way. However, I think my original implementation had the downside of overriding the `:if`, and I think the user would want to keep OptionalInclude's `:if` and still add their own.

Now we just need to think how to specify these options. @plexus, any ideas?